common/src/main/java/org/esbtools/message/admin/common/EsbMessageAdminServiceImpl.java

```diff
@@ -196,6 +210,80 @@ public void create(EsbMessage esbMessage, Map<String, List<String>> extractedHea
         entityMgr.persist(eme);
     }
 
+    public MetadataResponse resubmitMessage(EsbMessage esbMessage) {
+
+        ResubmitRequest request = new ResubmitRequest();
+
+        EsbMessageEntity persistedMessage = entityMgr.find(EsbMessageEntity.class, esbMessage.getId());
+
+        request.setHeaders( reduceToEsbHeaders(persistedMessage) );
+        request.setDestination( persistedMessage.getHeader("esbResubmitDestination").getValue() );
+        request.setSystem( persistedMessage.getSourceSystem() );
+
+        // scold the user for trying to update instead of insert
+        if( persistedMessage == null ){
+            throw new IllegalArgumentException("Message {\n}"+esbMessage.toString()+"\n} does not exist in backend store, cannot update.");
+        }
+
+        // explicitly check if the loaded message is in our list of allowed message types
+        if( isEditableMessage(esbMessage) ){
+            request.setPayload( esbMessage.getPayload() );
```

Comment:
We need to (un-encrypt and) inject any sensitive information previously extracted from the message (like on persist) before resubmitting.